### PR TITLE
Support for custom roles

### LINKED ISSUE #7625: Support for custom roles
Add support for custom roles in people management

### PR DESCRIPTION
**Pull request by koke**
This adds support for custom roles on Jetpack sites. Instead of using a hardcoded list of roles, we're now retrieving them from the API and storing them in Core Data.

Fixes #7625 

To test:

An easy way to add custom roles to a self hosted is to install WooCommerce, since it adds a `Customer` and a `Shop Manager` role.

- Test that you can view Users and they have the proper role
- Test that you can view Followers/Viewers
- Invite a user to view/follow 
- Invite a user with a custom role

Needs review: @jleandroperez

### CODE REVIEW

**Comment by jleandroperez** [comment]
Love this PR!!. I'm having a small glitch. Details Below:

### Bug: Can't Invite with Custom Roles
1. Setup a sandbox with Woo + Jetpack
2. Log into your dotcom account
3. Open the new blog and attempt to send an invite

Problem: The role `shop_manager` does not pass the **PeopleServiceRemote.validateInvitation** call. Error pasted below:

 ```Error Domain=WordPressKit.WordPressComRestApiError Code=0 "Invalid user role. Please try again." UserInfo={WordPressComRestApiErrorCodeKey=invalid_input, WordPressComRestApiErrorMessageKey=Invalid user role. Please try again., NSLocalizedDescription=Invalid user role. Please try again.}```

**Comment by koke** [comment]
> Invalid user role

I see this is an API bug that has been reported internally. The Android app had the same issue.

File: WordPress/Classes/Services/RoleService.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    /// Forces a refresh of roles from the api and stores them in Core Data.
+    ///
+    func fetchRoles(success: @escaping ([Role]) -> Void, failure: @escaping (Error) -> Void) {
+        remote.getUserRoles(siteID,
+                            success: { (remoteRoles) in
+                                let roles = self.mergeRoles(remoteRoles)
+                                success(roles)
+        },
+                            failure: failure)
+    }
```
Comment (jleandroperez) [inline WordPress/Classes/Services/RoleService.swift]: Minor comment: broken alignment(s) here!

File: WordPress/Classes/Services/RoleService.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+private extension RoleService {
+    func mergeRoles(_ remoteRoles: [RemoteRole]) -> [Role] {
+        let existingRoles = blog.roles ?? []
+        var rolesToKeep = [Role]()
+        var order = 0
+        for remoteRole in remoteRoles {
```
Comment (jleandroperez) [inline WordPress/Classes/Services/RoleService.swift]: Tip tip! (you'll love this one):

```
for (order, remoteRole) in remoteRoles.enumerated() {
```

Comment (koke) [inline WordPress/Classes/Services/RoleService.swift]: Ah, I knew there was something like that but couldn't remember the syntax

**Pull request merged**